test/unit/sql/data-types.test.ts

```diff
@@ -0,0 +1,1492 @@
+          expect: {
+            default: zeroFillUnsupportedError,
+            'mysql mariadb': 'SMALLINT UNSIGNED ZEROFILL',
+          },
+        },
+        {
+          title: 'SMALLINT(4).UNSIGNED.ZEROFILL',
+          dataType: DataTypes.SMALLINT(4).UNSIGNED.ZEROFILL,
+          expect: {
+            default: zeroFillUnsupportedError,
+            'mysql mariadb': 'SMALLINT(4) UNSIGNED ZEROFILL',
+          },
+        },
+        {
+          title: 'SMALLINT.ZEROFILL',
+          dataType: DataTypes.SMALLINT.ZEROFILL,
+          expect: {
+            default: zeroFillUnsupportedError,
+            'mysql mariadb': 'SMALLINT ZEROFILL',
+          },
+        },
+        {
+          title: 'SMALLINT(4).ZEROFILL',
+          dataType: DataTypes.SMALLINT(4).ZEROFILL,
+          expect: {
+            default: zeroFillUnsupportedError,
+            'mysql mariadb': 'SMALLINT(4) ZEROFILL',
+          },
+        },
+        {
+          title: 'SMALLINT.ZEROFILL.UNSIGNED',
+          dataType: DataTypes.SMALLINT.ZEROFILL.UNSIGNED,
+          expect: {
+            default: zeroFillUnsupportedError,
+            'mysql mariadb': 'SMALLINT UNSIGNED ZEROFILL',
+          },
+        },
+        {
+          title: 'SMALLINT(4).ZEROFILL.UNSIGNED',
+          dataType: DataTypes.SMALLINT(4).ZEROFILL.UNSIGNED,
+          expect: {
+            default: zeroFillUnsupportedError,
+            'mysql mariadb': 'SMALLINT(4) UNSIGNED ZEROFILL',
+          },
+        },
+      ];
+      for (const row of cases) {
+        testsql(row.title, row.dataType, row.expect);
+      }
+
+      describe('validate', () => {
+        it('should throw an error if `value` is invalid', () => {
+          const type = DataTypes.SMALLINT();
+
+          expect(() => {
+            type.validate('foobar');
+          }).to.throw(ValidationErrorItem, `'foobar' is not a valid smallint`);
+
+          expect(() => {
+            type.validate(123.45);
+          }).to.throw(ValidationErrorItem, '123.45 is not a valid smallint');
+        });
+
+        it('should not throw if `value` is an integer', () => {
+          const type = DataTypes.SMALLINT();
+
+          expect(() => type.validate(-32_768)).not.to.throw();
+          expect(() => type.validate('32767')).not.to.throw();
+        });
+      });
+    });
+
+    describe('MEDIUMINT', () => {
+      const zeroFillUnsupportedError = new Error(`${dialectName} does not support the MEDIUMINT.ZEROFILL data type.
+See https://sequelize.org/docs/v7/other-topics/other-data-types/ for a list of supported data types.`);
+
+      const cases = [
+        {
+          title: 'MEDIUMINT',
+          dataType: DataTypes.MEDIUMINT,
+          expect: {
+            'mariadb mysql sqlite': 'MEDIUMINT',
+            // falls back to larger type + CHECK constraint
+            'db2 ibmi mssql postgres snowflake': 'INTEGER',
+          },
+        },
+        {
+          title: 'MEDIUMINT(2)',
```

Comment:
Any reason this was updated from `MEDIUMINT(6)` to `MEDIUMINT(2)`?

Reply:
Probably copy-paste. But it doesn't really matter the number will be inlined as-is